src/audio/codec_adapter/codec_adapter.c

```diff
@@ -407,18 +423,19 @@ static int codec_adapter_copy(struct comp_dev *dev)
 		codec_adapter_copy_from_lib_to_sink(&codec->cpd, &local_buff->stream,
 						    codec->cpd.produced);
 
-		bytes_to_process -= codec->cpd.produced;
-		processed += codec->cpd.produced;
+		bytes_to_process -= codec->cpd.consumed;
+		processed += codec->cpd.consumed;
+		produced += codec->cpd.produced;
 	}
 
-	if (!processed && !cd->deep_buff_bytes) {
+	if (!produced && !cd->deep_buff_bytes) {
 		comp_dbg(dev, "codec_adapter_copy(): nothing processed in this call");
 		goto end;
-	} else if (!processed && cd->deep_buff_bytes) {
+	} else if (!produced && cd->deep_buff_bytes) {
 		goto db_verify;
 	}
 
-	audio_stream_produce(&local_buff->stream, processed);
+	audio_stream_produce(&local_buff->stream, produced);
 	comp_update_buffer_consume(source, processed);
```

Comment:
I see you've added produced variable to change processed, but still use both.
What will be the difference between produced and processed?

I think the intention was to change processed to produced everywhere, but you missed some places.

Reply:
`processed` is the number of bytes consumed from the input buffer. These are compressed bytes. (maybe we should rename it to consumed). 

`produced` is the number of bytes resulted after applying the decompression algorithm. Current existing implementation works when the input and output data are PCM samples (like Dolby Audio Processing using by @mrajwa for implementation). In this case the number of `consumed` data from source buffer is always equal with the number of bytes `produced` to the output buffer.